packages/ui/src/useStoredState.ts

```diff
@@ -0,0 +1,36 @@
+import {useEffect, useState} from "react";
+
+import {Unifier} from "./Unifier";
+
+export const useStoredState = <T>(
+  key: string,
+  initialValue?: T
+): [T | undefined | null, (value: T | undefined | null) => Promise<void>] => {
+  const [state, setState] = useState<T | undefined | null>(initialValue);
+
+  useEffect(() => {
+    // Function to fetch data from AsyncStorage
+    const fetchData = async (): Promise<void> => {
```

Comment:
What's the difference between using useCallback to define this function and defining the function within the useEffect? Given that useEffect will be called several times, would it be better to set it up as a useCallback so that it doesn't get redefined?

Reply:
I think @swanka might be on to something here. I found a good example of a library that does this (https://usehooks-ts.com/react-hook/use-local-storage) and it uses the `useCallback` way of defining the read data. We can't use that library directly since it's not for Native, but I think besides swapping out the method of interacting with local storage we can use it.